File: etamen-backend/app/Modules/Labs/Http/Requests/CreateLabOrderRequest.php
<?php

namespace App\Modules\Labs\Http\Requests;

use App\Modules\Labs\Domain\Enums\LabOrderItemType;
use App\Modules\Labs\Domain\Enums\LabSampleCollectionMethod;
use Illuminate\Foundation\Http\FormRequest;
use Illuminate\Validation\Rule;

class CreateLabOrderRequest extends FormRequest
{
    public function authorize(): bool
    {
        return true;
    }

    public function rules(): array
    {
        return [
            'lab_provider_id' => ['required', 'integer', 'exists:providers,id'],
            'sample_collection_method' => ['required', Rule::in(LabSampleCollectionMethod::values())],
            'collection_address' => ['required_if:sample_collection_method,'.LabSampleCollectionMethod::HomeCollection->value, 'nullable', 'string', 'max:1000'],
            'scheduled_at' => ['nullable', 'date', 'after:now'],
            'notes' => ['nullable', 'string', 'max:2000'],
            'items' => ['required', 'array', 'min:1'],
            'items.*.item_type' => ['required', Rule::in(LabOrderItemType::values())],
            'items.*.test_id' => ['nullable', 'integer', 'exists:lab_tests,id', 'required_if:items.*.item_type,'.LabOrderItemType::Test->value],
            'items.*.package_id' => ['nullable', 'integer', 'exists:lab_packages,id', 'required_if:items.*.item_type,'.LabOrderItemType::Package->value],
            'items.*.quantity' => ['required', 'integer', 'min:1', 'max:99'],
            'subtotal' => ['prohibited'],
            'discount_total' => ['prohibited'],
            'commission_amount' => ['prohibited'],
            'provider_net_amount' => ['prohibited'],
            'grand_total' => ['prohibited'],
            'currency' => ['prohibited'],
            'payment_status' => ['prohibited'],
            'order_status' => ['prohibited'],
            'paid_at' => ['prohibited'],
            'items.*.unit_price' => ['prohibited'],
            'items.*.line_total' => ['prohibited'],
        ];
    }
}

File: etamen-flutter/lib/features/labs/data/models/create_lab_order_request.dart
import 'package:etamen_app/features/labs/domain/entities/lab_cart_item.dart';

enum LabSampleCollectionMethod {
  branchVisit('branch_visit'),
  homeCollection('home_collection');

  const LabSampleCollectionMethod(this.wireValue);

  final String wireValue;
}

class CreateLabOrderRequest {
  const CreateLabOrderRequest({
    required this.labProviderId,
    required this.items,
    required this.sampleCollectionMethod,
    this.collectionAddress,
    this.notes,
  });

  final int labProviderId;
  final List<LabCartItem> items;
  final LabSampleCollectionMethod sampleCollectionMethod;
  final String? collectionAddress;
  final String? notes;

  Map<String, dynamic> toJson() {
    return {
      'lab_provider_id': labProviderId,
      'sample_collection_method': sampleCollectionMethod.wireValue,
      if (collectionAddress?.trim().isNotEmpty == true)
        'collection_address': collectionAddress!.trim(),
      if (notes?.trim().isNotEmpty == true) 'notes': notes!.trim(),
      'items': items
          .map(
            (item) => {
              'item_type': item.type == LabCartItemType.test
                  ? 'test'
                  : 'package',
              if (item.type == LabCartItemType.test) 'test_id': item.itemId,
              if (item.type == LabCartItemType.package)
                'package_id': item.itemId,
              'quantity': item.quantity,
            },
          )
          .toList(growable: false),
    };
  }
}
